File: Codeforces with Cpp/binary search.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int n,i,first,last,mid,b;
    scanf("%d",&n);
    int a[n];


    for(i=0;i<n;i++)
    {
        scanf("%d",&a[i]);
    }

    scanf("%d",&b);

    first=0;
    last=n-1;

    while(first<=last)
    {
        mid=(first+last)/2;

        if(a[mid]==b)
        {
            printf("%d",mid);
            break;
        }
        else if(a[mid]<b)
        {
            first=mid+1;
        }
        else if(a[mid]>b)
        {
            last=mid-1;
        }
    }




 return 0;

}

File: Data Structure & Algorithm/Sorting & Searching/Binary Search.py
a = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19]

first=0;last=len(a)-1

n = int(input())
result=-1
while first<=last:
    mid=(first+last)//2

    if a[mid]== n:
        result=mid
        break
    elif a[mid]<n:
        first=mid+1
    else:
        last=mid-1

    print (first,last,mid)


if result==-1:
    print('Item not found')
else:
    print('The index is:',result)
